util/topgen/resets.py

```diff
@@ -214,3 +214,29 @@ def has_shadowed_reset(self) -> bool:
                 return True
 
         return False
+
+
+class UnmanagedReset:
+    '''An unmanaged reset (input to the top-level).'''
+
+    def __init__(self, raw: Dict[str, object]):
+        self.name = str(raw['name'])
```

Comment:
Practically speaking, we're currently expecting the dict to be a singleton that just has a `name` key. We should probably check for that here.

Reply:
I added a check.